### PR TITLE
Add support for `ATTACH PARTITION ALL`

### LINKED ISSUE #59376: `ATTACH PARTITION ALL` should be supported
**Use case**

```
milovidov-desktop :) ALTER TABLE hits2 ATTACH PARTITION all FROM hits

ALTER TABLE hits2
    ATTACH PARTITION ALL FROM hits

Query id: aebb4b9c-8413-4213-97d2-e2c662aecbdd


Elapsed: 0.008 sec. 

Received exception from server (version 24.1.1):
Code: 344. DB::Exception: Received from localhost:9000. DB::Exception: Only support DROP/DETACH PARTITION ALL currently. (SUPPORT_IS_DISABLED)
```

It should attach all the parts, regardless of the partition key.

### PR DESCRIPTION
**Pull request by allmazz**
### Changelog category (leave one):
- New Feature

Closes https://github.com/ClickHouse/ClickHouse/issues/59376
Example usage: `ALTER TABLE data ATTACH PARTITION ALL`

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add support for `ATTACH PARTITION ALL`

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 43932528800d4c9b22b266aeb502d9a2de0b8730 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61107/43932528800d4c9b22b266aeb502d9a2de0b8730/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by pufit — APPROVED** [review]

File: tests/queries/0_stateless/00753_alter_attach.sql [hunk 1]
```diff
@@ -90,3 +90,45 @@ OPTIMIZE TABLE partition_all2 PARTITION ALL; -- { serverError 344 }
...
+ALTER TABLE partition_attach_all DROP PARTITION ALL;
+SELECT * FROM partition_attach_all ORDER BY x;
+ALTER TABLE partition_attach_all ATTACH PARTITION ALL;
+SELECT * FROM partition_attach_all ORDER BY x;
+
+CREATE TABLE replicated_partition_attach_all (x UInt64, p UInt8)
+    ENGINE = ReplicatedMergeTree('/clickhouse/tables/test_00753_{database}/replicated_partition_attach_all', '1')
+        ORDER BY x
+        PARTITION BY p;
+
```
Comment (pufit) [inline tests/queries/0_stateless/00753_alter_attach.sql:117]: Let's insert some data in the replicated table too.

Comment (allmazz) [reply tests/queries/0_stateless/00753_alter_attach.sql:118]: thanks, it's done. idk how i missed that

**Comment by alexey-milovidov** [comment]
@allmazz, It does not work:

```
milovidov-desktop :) ALTER TABLE uk_price_paid_local ATTACH PARTITION ALL FROM uk_price_paid_web

ALTER TABLE uk_price_paid_local
    (ATTACH PARTITION ALL FROM uk_price_paid_web)

Query id: 94d3716e-f88a-40f3-8f53-ffd1f1cc3bb4


Elapsed: 0.013 sec. 

Received exception from server (version 24.3.1):
Code: 344. DB::Exception: Received from localhost:9000. DB::Exception: Only support DROP/DETACH/ATTACH PARTITION ALL currently. (SUPPORT_IS_DISABLED)

```

**Comment by allmazz** [comment]
> @allmazz, It does not work:
> 
> ```
> milovidov-desktop :) ALTER TABLE uk_price_paid_local ATTACH PARTITION ALL FROM uk_price_paid_web
> 
> ALTER TABLE uk_price_paid_local
>     (ATTACH PARTITION ALL FROM uk_price_paid_web)
> 
> Query id: 94d3716e-f88a-40f3-8f53-ffd1f1cc3bb4
> 
> 
> Elapsed: 0.013 sec. 
> 
> Received exception from server (version 24.3.1):
> Code: 344. DB::Exception: Received from localhost:9000. DB::Exception: Only support DROP/DETACH/ATTACH PARTITION ALL currently. (SUPPORT_IS_DISABLED)
> 
> ```

I will look into this

**Comment by allmazz** [comment]
@alexey-milovidov it does not work in last stable release too

**Comment by alexey-milovidov** [comment]
Yes, but it is my expectation from the feature, introduced in this pull request.
The implementation should be complete and consistent.

**Comment by allmazz** [comment]
> Yes, but it is my expectation from the feature, introduced in this pull request. The implementation should be complete and consistent.

certainly, i will submit a fix today

**Comment by allmazz** [comment]
> Yes, but it is my expectation from the feature, introduced in this pull request. The implementation should be complete and consistent.

There is a problem: we can't perform this operation atomically in replicated tables like we can with common, so should we support this in replicated?

**Comment by allmazz** [comment]
> Yes, but it is my expectation from the feature, introduced in this pull request.
> The implementation should be complete and consistent.

There is a problem: we can't perform this operation atomically on replicated tables without a lot of changes, so should we support this on them?

**Comment by alexey-milovidov** [comment]
It is ok to do it for each part atomically, or even do it atomically for each part from the user's standpoint but leaving garbage in case of a failure.

**Comment by allmazz** [comment]
> It is ok to do it for each part atomically, or even do it atomically for each part from the user's standpoint but leaving garbage in case of a failure.

https://github.com/ClickHouse/ClickHouse/pull/61987

**Comment by jiayeZhu** [comment]
Hi, is it possible to also support `REPLACE PARTITION ALL FROM TABLE`? thanks

**Comment by allmazz** [comment]
> Hi, is it possible to also support `REPLACE PARTITION ALL FROM TABLE`? thanks

yes, i did it, check pr above

**Pull request merged**